src/main/java/com/epam/training/model/AbstractChristmasGift.java

```diff
@@ -0,0 +1,47 @@
+package com.epam.training.model;
+
+import java.util.Objects;
+
+public abstract class AbstractChristmasGift {
```

Comment:
AbstractCandy (not a gift). Gift is not a part of common hierarchy, Gift is NOT a candy

Reply:
@FilippStankevichTR Done.
Please, see the changes here: https://github.com/AIniuseva/epam-oop-task/pull/1/files#diff-2a64712a420d8cd7cec94191cd33986efb976571d2745114e4c6868dff24f2b8R7